### PR TITLE
MSC4170: 403 error responses for profile APIs

### PR DESCRIPTION
**Pull request by Johennes**
Spec for https://github.com/matrix-org/matrix-spec-proposals/pull/4170.

### Pull Request Checklist

* [x] Pull request includes a [changelog file](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#adding-to-the-changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#sign-off)
* [ ] Pull request is classified as ['other changes'](https://github.com/matrix-org/matrix-spec/blob/master/CONTRIBUTING.rst#other-changes)

Preview: https://pr1867--matrix-spec-previews.netlify.app

### CODE REVIEW

File: data/api/client-server/profile.yaml
Comment (Johennes) [inline data/api/client-server/profile.yaml:102]: I don't know if this is correct but technically we're clarifying the 403 response introduced in Matrix v1.2.

**Review by richvdh — APPROVED** [review]
LGTM, but will await opinions from other SCT members on https://github.com/matrix-org/matrix-spec/pull/1867/files#r1636536644

File: data/api/server-server/query.yaml
Comment (richvdh) [inline data/api/server-server/query.yaml:186]: actually: given MSC3550 didn't mention the federation API at all, making this change without an MSC seems like a bit of a stretch.

**Review by richvdh — CHANGES_REQUESTED** [review]

File: data/api/server-server/query.yaml
Comment (richvdh) [reply data/api/server-server/query.yaml:186]: (and while we're at it maybe we should discuss the additional client endpoints too, since they weren't *technically* covered by the MSC or the original spec PR)

Comment (clokep) [reply data/api/server-server/query.yaml:186]: > given MSC3550 didn't mention the federation API at all

It mentions that the reason the MSC is being written is for requests over federation, but then doesn't mention the federation endpoints at all.

I'm a bit torn on it.

Comment (Johennes) [reply data/api/server-server/query.yaml:186]: I am (not eager but definitely) willing to put up a short MSC for this. I would like to ensure that it's time well spent though. So I'll hold off of doing anything until you've found an agreement on whether an MSC is needed or not.

Comment (turt2live) [reply data/api/server-server/query.yaml:186]: We discussed it briefly at the time of richvdh's comment, and I think the thread here captures that discussion well. It feels like it needs an MSC because it's relatively unclear what the original MSC's intended scope was. 

As a formality, my vote is +1.0 to needing an MSC because the intention and history are different stories.

Comment (Johennes) [reply data/api/server-server/query.yaml:186]: Ok, makes sense. I'll try and put something together next week.

Comment (Johennes) [reply data/api/server-server/query.yaml:186]: This PR is now blocked on https://github.com/matrix-org/matrix-spec-proposals/pull/4170

Comment (Johennes) [reply data/api/server-server/query.yaml]: This PR has now been updated as per the accepted MSC.

**Comment by Johennes** [comment]
I've optimistically used `x-addedInMatrixVersion: "1.12"` but this doesn't block the 1.12 release in any way. If the release happens before this PR lands, I'll bump the versions to 1.13.

**Review by richvdh — APPROVED** [review]
LGTM. THanks!

**Pull request merged**
